staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -226,9 +243,16 @@ func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
 
 			start := strings.LastIndex(js[:syntax.Offset], "\n") + 1
 			line := strings.Count(js[:start], "\n")
-			return fmt.Errorf("json: line %d: %s", line, syntax.Error())
+			return JSONSyntaxError{
+				Line: line,
+				Err:  fmt.Errorf(syntax.Error()),
+			}
 		}
 	}
+
+	if err != nil && err != io.EOF && hasPrefix([]byte(err.Error()), []byte("yaml: line")) {
```

Comment:
This is less fine.  `strings.HasPrefix`?  Also, is there any actual evidence this is consistent or guaranteed?